src/echoclient/EchoEvent.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace echoclient;
+
+class EchoEvent implements \JsonSerializable
+{
+
+    private $class;
+    private $source;
+    private $props;
+    private $userId;
+    private $timestamp;
+
+    /**
+     * @param mixed $dependencies
+     */
+    public function __construct(
+        $class,
+        $source,
+        $props = array(),
+        $userId = null,
+        $timestamp = null
+    ) {
+        if (!defined('ECHO_CLASS_PREFIX')) {
+            define('ECHO_CLASS_PREFIX', '');
+        }
+
+        if (!is_string($class)) {
+            throw new \InvalidArgumentException("`class` must be a string");
+        }
+
+        if (!is_string($source)) {
+            throw new \InvalidArgumentException("`source` must be a string");
+        }
+
+        if (!is_array($props)) {
+            throw new \InvalidArgumentException("`props` must be an array");
+        }
+
+        if (!is_null($userId) && !is_string($userId)) {
+            throw new \InvalidArgumentException("`userId` must be a string");
+        }
+
+        if (!is_null($timestamp) && !is_string($timestamp)) {
+            throw new \InvalidArgumentException("`timestamp` must be a string");
```

Comment:
I'm still not 100% sure what is acceptable here.  The old APIs say [pass a timestamp as an integer (seconds since UTC)](https://talisecho.docs.apiary.io/#reference/events/events/add-events); the [new APIs say ISO-8601 date-time string](https://analytics.talis.com/docs/#/Events/eventsPost).  The code [defaults to an integer](https://github.com/talis/echo-server/blob/master/app/models/processor.js#L62) and stores as an `ISODate`.  Mongoose seems to configure with [a schema expecting a date](https://github.com/talis/echo-server/blob/e3f1dd29ba46ede7a5e2cbfb2b1fbc64d9f5444d/app/models/event.js#L9-L12).

In short: I don't know what's right here.  Can I get confirmation from someone who does know?

Reply:
I spoke to @markwallsgrove he advised sending it as a string.

Reply:
Cool, thanks. I’ve attempted to flip RL accordingly, though it has timezone issues.